cpp/CMakeLists.txt

```diff
@@ -246,46 +250,64 @@ file(INSTALL ${LIBCXX_INCLUDE_DIR}     DESTINATION ${CMAKE_BINARY_DIR}/include/l
 ###################################################################################################
 # - find zlib -------------------------------------------------------------------------------------
 
-find_package(ZLIB REQUIRED)
+CPMFindPackage(
+  NAME ZLIB
+  GITHUB_REPOSITORY madler/zlib
+  VERSION 1.2.11
+)
 
-message(STATUS "ZLIB: ZLIB_LIBRARIES set to ${ZLIB_LIBRARIES}")
-message(STATUS "ZLIB: ZLIB_INCLUDE_DIRS set to ${ZLIB_INCLUDE_DIRS}")
+message(STATUS "ZLIB: ZLIB_LIBRARY set to ${ZLIB_LIBRARY}")
+message(STATUS "ZLIB: ZLIB_INCLUDE_DIR set to ${ZLIB_INCLUDE_DIR}")
 
-if(ZLIB_FOUND)
-    message(STATUS "ZLib found in ${ZLIB_INCLUDE_DIRS}")
+if(ZLIB_INCLUDE_DIR)
+    message(STATUS "ZLib found in ${ZLIB_INCLUDE_DIR}")
 else()
     message(FATAL_ERROR "ZLib not found, please check your settings.")
-endif(ZLIB_FOUND)
+endif(ZLIB_INCLUDE_DIR)
+
 
 ###################################################################################################
 # - find boost ------------------------------------------------------------------------------------
 
 # Don't look for a CMake configuration file
 set(Boost_NO_BOOST_CMAKE ON)
 
-find_package(
-    Boost REQUIRED MODULE
-    COMPONENTS filesystem
+CPMFindPackage(
+  NAME Boost
+  GITHUB_REPOSITORY Orphis/boost-cmake
+  VERSION 1.67.0
```

Comment:
This should be version matched with the conda recipe for libcudf

Reply:
This is a weird one. People prefer `Orphis` or `Element-0` for the repos for this. There is very little difference between the branches. No one actively maintains. I feel like both are a weird and terrifying attack surface. There are a few others but they are not as widely updated or used. No one is tagging for 1.70+ (which we would need, 1.73+ I believe) `master` on both of the primary `boost-cmake` providers is update to `1.70` with some additions that maintain further compatibility but I cannot verify the completeness in the scope of this thing.

Do we just say "Install the correct one or fail", do we start maintaining our own, or do we grab master using `GIT_TAG` and hope for the best?

Reply:
I think we'd be okay to not version match here and given our usage of boost is so tiny we could be more lenient on versioning for now. Given we'd also want to support `apt` / `yum` installs which would likely lag conda as well. I'd say we should be able to use 1.70 for now given the boost-cmake repo supports it without issue, but we should move to 1.72.0 when the boost-cmake repo supports it.

Looks like upgrading the boost version in the boost-cmake repo is relatively straightforward, i.e. for 1.70 upgrade: https://github.com/Orphis/boost-cmake/commit/60fb5faea7db755757ca9b062e754658d17ee11e